lib/ansible/modules/user.py

```diff
@@ -3235,14 +3284,56 @@ def modify_user(self):
                             self.module.fail_json(name=self.name, msg=err, rc=rc)
 
         # Manage password
-        if self.update_password == 'always' and self.password is not None and info[1] != self.password:
-            cmd = [self.module.get_bin_path('chpasswd', True)]
-            cmd.append('--encrypted')
-            data = '{name}:{password}'.format(name=self.name, password=self.password)
-            rc, out, err = self.execute_command(cmd, data=data)
+        current_password = to_native(user_info[1])
+        new_password = self._build_password_string(current_password)
+        if self.update_password == 'always':
+            lock_status_mismatch = self.password_lock and not current_password.startswith('!')
```

Comment:
With the earlier comment, I'd only expect this to be updated too:
```suggestion
            lock_status_mismatch = self.password_lock and not current_password.startswith(LOCK_INDICATOR)
```

Reply:
I'll work on this tomorrow.

Reply:
Changed!